src/raftstore/store/debug.rs

```diff
@@ -67,6 +68,44 @@ impl Debugger {
         }
     }
 
+    pub fn region_info(
+        &self,
+        region_id: u64,
+    ) -> Result<
+        (
+            Option<raft_serverpb::RaftLocalState>,
+            Option<raft_serverpb::RaftApplyState>,
+            Option<raft_serverpb::RegionLocalState>,
+        ),
+    > {
+        let raft_state_key = keys::raft_state_key(region_id);
+        let raft_state = self.engines
+            .raft_engine
+            .get_msg::<raft_serverpb::RaftLocalState>(&raft_state_key);
+
+        let apply_state_key = keys::apply_state_key(region_id);
+        let apply_state = self.engines
+            .kv_engine
+            .get_msg_cf::<raft_serverpb::RaftApplyState>(CF_RAFT, &apply_state_key);
+
+        let region_state_key = keys::region_state_key(region_id);
+        let region_state = self.engines
+            .kv_engine
+            .get_msg_cf::<raft_serverpb::RegionLocalState>(CF_RAFT, &region_state_key);
+
+        match (raft_state, apply_state, region_state) {
+            (Err(e), Err(_), Err(_)) => Err(box_err!(e)),
```

Comment:
What if it's `Ok()`, `Err(_)`, `Err(_)`

Reply:
Since it is for debugging, I think it should return region info on a best effort basis.

Reply:
No, should report error instead. Otherwise we can't know precisely that the data is damaged, or written in the wrong way or just some arbitrarily rocksdb errors.